modules/configuration/src/test/java/org/apache/ignite/internal/configuration/util/OrderedMapTest.java

```diff
@@ -0,0 +1,151 @@
+
+    /** Tests put/get/remove consistency on a single key. */
+    @Test
+    public void putGetRemove() {
+        assertNull(map.get("key1"));
+
+        map.put("key", "value");
+
+        assertEquals(1, map.size());
+        assertEquals("value", map.get("key"));
+
+        map.remove("key");
+
+        assertNull(map.get("key1"));
+    }
+
+    /** Tests that {@link OrderedMap#put(String, Object)} preserves order. */
+    @Test
+    public void keysOrder() {
+        map.put("key1", "value1");
+        map.put("key2", "value2");
+
+        assertEquals(2, map.size());
+        assertEquals(List.of("key1", "key2"), map.keys());
+
+        map.clear();
+
+        assertEquals(0, map.size());
+        assertEquals(List.of(), map.keys());
+
+        map.put("key2", "value2");
+        map.put("key1", "value1");
+
+        assertEquals(2, map.size());
+        assertEquals(List.of("key2", "key1"), map.keys());
+
+        map.put("key2", "value3");
+        assertEquals(List.of("key2", "key1"), map.keys());
+    }
+
+    /** Tests that {@link OrderedMap#putByIndex(int, String, Object)} preserves order. */
+    @Test
+    public void putByIndex() {
+        map.putByIndex(0, "key1", "value1");
+
+        map.putByIndex(0, "key2", "value2");
+
+        assertEquals(List.of("key2", "key1"), map.keys());
+
+        map.putByIndex(1, "key3", "value3");
+
+        assertEquals(List.of("key2", "key3", "key1"), map.keys());
+
+        map.putByIndex(3, "key4", "value4");
+
+        assertEquals(List.of("key2", "key3", "key1", "key4"), map.keys());
+
+        map.putByIndex(5, "key5", "value5");
+
+        assertEquals(List.of("key2", "key3", "key1", "key4", "key5"), map.keys());
+    }
+
+    /** Tests that {@link OrderedMap#putAfter(String, String, Object)} preserves order. */
+    @Test
+    public void putAfter() {
+        map.putAfter("foo", "key1", "value1");
+
+        assertEquals(List.of("key1"), map.keys());
+
+        map.putAfter("key1", "key2", "value2");
+
+        assertEquals(List.of("key1", "key2"), map.keys());
+
+        map.putAfter("key1", "key3", "value3");
+
+        assertEquals(List.of("key1", "key3", "key2"), map.keys());
+
+        map.putAfter("key2", "key4", "value4");
+
+        assertEquals(List.of("key1", "key3", "key2", "key4"), map.keys());
+
+        map.putAfter("foo", "key5", "value5");
+
+        assertEquals(List.of("key1", "key3", "key2", "key4", "key5"), map.keys());
+    }
+
+    /** Tests basic invariants of {@link OrderedMap#rename(String, String)} method. */
+    @Test
+    public void rename() {
+        map.put("key1", "value1");
+        map.put("key2", "value2");
+        map.put("key3", "value3");
+
+        map.rename("key2", "key4");
+
+        assertEquals("value2", map.get("key4"));
+
+        assertEquals(List.of("key1", "key4", "key3"), map.keys());
+    }
+
+    /** Tests that {@link OrderedMap#reorderKeys(List)} reorders keys properly. */
+    @Test
+    public void reorderKeys() {
```

Comment:
I'll test error cases in NamedListOrderTest. OrderMap class lacks some checks because they are duplicated. I made OrderMap package-private so that it won't be accidentally reused.